### DISCUSSION TITLE
Automerge Octoclairvoyant repository non functional

### DISCUSSION THREAD

**Discussion by HonkingGoose** [renovatebot/renovate#10554] [Request Help] [answered]
### Problem:

I've setup a basic automerge config, which I think should work, but Renovate bot does not automerge a lockfile update after waiting 5 hours, while all tests pass.

I'm not sure why it's not working properly.... Did I make a Renovate config error, is the Vercel deployment not passed to the Renovate bot, is PR approval a problem when automerging, or something else?

### Desired end-goal:

1. Create branch for certain updates (lockfile, prettier, ESLint)
2. Run tests on the branch (GitHub tests, Vercel deployment)
3. Merge into `main` if all tests pass

### Things I've checked

The Renovate branch does run all 3 tests (2 GitHub tests, 1 Vercel deployment).
The status is green on GitHub for the branch.
I've waited 5 hours to see if Renovate would automerge.

Link to lockfile update branch: https://github.com/Belco90/octoclairvoyant/tree/renovate/lock-file-maintenance
The tests are passing on the latest commit on this branch.

### Current Renovate config

```json
{
  "extends": ["config:base"],
  "dependencyDashboard": true,
  "prCreation": "not-pending",
  "lockFileMaintenance": {
    "enabled": true,
    "automerge": true
  },
  "automergeType": "branch",
  "internalChecksFilter": "strict",
  "packageRules": [
    {
      "description": "Wait 3 days before creating a npm update PR",
      "matchDatasources": ["npm"],
      "stabilityDays": 3
    },
    {
      "description": "Automerge ESLint and Prettier updates",
      "matchDepTypes": ["devDependencies"],
      "matchPackagePatterns": ["eslint", "prettier"],
      "automerge": true
    }
  ]
}
```

### Current test setup

GitHub workflow files: https://github.com/Belco90/octoclairvoyant/tree/main/.github/workflows

### Link to repository

https://github.com/Belco90/octoclairvoyant

**Comment by viceice** [comment, marked as answer]
@HonkingGoose The debug logs should mention why automerge doesn't work.

If you have required approvals you need to use pr merge as a branch can't be approved.

**Reply by HonkingGoose** [reply]
> If you have required approvals you need to use pr merge as a branch can't be approved.

Ooh I think that might be the problem, we require PR approval before merging... 😄 

I though that the "branch automerge" method skips the "PR approval" branch protection... 🙈

**Reply by rarkins** [reply]
Requiring PR approval implicitly requires PRs too.

**Reply by HonkingGoose** [reply]
> Requiring PR approval implicitly requires PRs too.

Makes sense... 😄
